src/main/kotlin/no/skatteetaten/aurora/boober/model/AuroraConfig.kt

```diff
@@ -90,15 +95,20 @@ data class AuroraConfig(val auroraConfigFiles: List<AuroraConfigFile>, val affil
 
     fun patchFile(filename: String, jsonPatchOp: String, previousVersion: String? = null): Pair<AuroraConfigFile, AuroraConfig> {
 
-        val mapper = jacksonYamlObjectMapper()
-        val patch: JsonPatch = mapper.readValue(jsonPatchOp, JsonPatch::class.java)
+        val patch: JsonPatch = yamlMapper.readValue(jsonPatchOp, JsonPatch::class.java)
 
         val auroraConfigFile = findFile(filename)
                 ?: throw IllegalArgumentException("No such file $filename in AuroraConfig ${affiliation}")
-        val originalContentsNode = mapper.convertValue(auroraConfigFile.contents, JsonNode::class.java)
 
-        val fileContents = patch.apply(originalContentsNode)
-        return updateFile(filename, fileContents, previousVersion)
+        val fileContents = patch.apply(auroraConfigFile.asJsonNode)
+
+        val writeMapper = if (filename.endsWith(".yaml")) {
+            yamlMapper
+        } else jsonMapper
+
+        val rawContents = writeMapper.writerWithDefaultPrettyPrinter().writeValueAsString(fileContents)
+        //TODO how do we handle this with regards to yaml/json.
```

Comment:
This is really the only place where I am wondering if the comments in the file will disappear. How does jackson handle comments in json/yaml files. Are they just ignored or are they a seperate type?

If they are just ignored then they will disappear here.

Reply:
Do we need to pretty print here? If we just use the fileContents directly we don't have to mess about with the mappers in this class at all.